frontend/src/pages/pipelines/global/modelCustomization/baseModelSection/InlineEditText.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import {
+  TextInput,
+  Button,
+  Flex,
+  FlexItem,
+  Content,
+  Alert,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { PencilAltIcon, CheckIcon, TimesIcon } from '@patternfly/react-icons';
+
+interface InlineEditTextProps {
+  text: string;
+  onSave: (text: string) => void;
+  checkSupported: (text: string) => boolean;
+  unsupportedMessage: string;
+}
+
+const InlineEditText: React.FC<InlineEditTextProps> = ({
+  text,
+  onSave,
+  checkSupported,
+  unsupportedMessage,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [inputValue, setInputValue] = useState(text);
+  const [showUnsupportedMessage, setShowUnsupportedMessage] = useState(false);
+
+  const isEmpty = React.useMemo(() => inputValue.trim().length === 0, [inputValue]);
+
+  React.useEffect(() => {
+    setShowUnsupportedMessage(!(isEditing || isEmpty || checkSupported(inputValue)));
+  }, [isEditing, inputValue, isEmpty, checkSupported]);
+
+  return (
+    <Stack>
+      <StackItem>
+        <Flex alignItems={{ default: 'alignItemsCenter' }}>
+          {isEditing ? (
+            <>
+              <FlexItem grow={{ default: 'grow' }} style={{ maxWidth: 500 }}>
+                <TextInput
+                  aria-label="Edit text"
+                  value={inputValue}
+                  onChange={(_, value) => setInputValue(value)}
+                  autoFocus
+                  style={{ maxWidth: '100%' }}
+                />
```

Comment:
If i enter after adding text, the page gets refresh.

Suggested fix: 
```suggestion
                <TextInput
                  aria-label="Edit text"
                  value={inputValue}
                  onChange={(_, value) => setInputValue(value)}
                  autoFocus
                  style={{ maxWidth: '100%' }}
                  onKeyDown={(e) => {
                    if (e.key === 'Enter') {
                      e.preventDefault(); // Prevent form submission or page refresh
                      onSave(inputValue);
                      setIsEditing(false);
                    }
                  }}
                />
```

Reply:
Thanks! In that case, I'll handle Escape too.